Add recipe for pyunlocbox

mdeff opened the pull request.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/pyunlocbox```) and found it was in an excellent condition.

synapticarbors commented on recipes/pyunlocbox/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+  build:
+    - python
+    - setuptools
+    - numpy
```
It looks like `numpy` and `scipy` are not `build` requirements since they are not imported by the `setup.py` directly or by side-effect. The `--single-version-externally-managed` allows you to ignore what is in `install_requires` for the `build` reqs.

mdeff replied:
Thanks, will remove `numpy` and `scipy`.

synapticarbors commented on recipes/pyunlocbox/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+test:
+  imports:
+    - pyunlocbox
+    - pyunlocbox.tests
```
No need to import the `tests` submodule.

mdeff replied:
Okay!

synapticarbors commented on recipes/pyunlocbox/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+about:
+  home: https://github.com/epfl-lts2/pyunlocbox
+  license: BSD-3-Clause
+  license_family: BSD
```
Please manually package the `license_file` and raise an issue in the source repo asking for it to be included in future source distributions. Add a link to the issue as a comment in the recipe:

https://conda-forge.org/docs/meta.html#packaging-the-license-manually

mdeff replied:
I'm managing the source repo and thought about including the license file in the next release. But I'd rather not make a release just for that. As I think the BSD does not require a license file in the package, would it be acceptable to have none until the next release?

mdeff commented:
I'm considering running the test suite for this package. Do you think it's worth it? Knowing it's pure Python and it's already checked by travis-ci on the source repo, but only for Linux. If yes, where should I put the test dependencies and test run command? Thanks

mdeff commented:
[Travis passed on my fork](https://travis-ci.org/mdeff/staged-recipes/builds/316926735)

mdeff commented:
@conda-forge/staged-recipes can we merge? I've addressed all the comments from @synapticarbors, made a new PyPI release and all the tests are passing.

synapticarbors commented:
@mdeff -- Thanks for your contribution to conda-forge. A feedstock for this recipe should be generated soon and you'll be receiving several automated messages about it. When the feedstock has been rendered, please take a look at the README in that repo for information about updating and maintaining the recipe. One of the key points is all updates should be done by forking that repo and following a standard pull request workflow. If you haven't looked at it already the docs are quite helpful:

https://conda-forge.org/docs/

If you run into any difficulties please let us know.

mdeff commented:
Thanks for the review and info @synapticarbors!
